### DISCUSSION TITLE
Installation on Windows

### DISCUSSION THREAD

**Discussion by OscarLahaie** [ocamllabs/vscode-ocaml-platform#944] [Q&A] [answered]
I run vscode on windows and i wonder if i can have the same features from Cygwin as on WSL. 
For example the ability to see/change of switch directly in vscode is not working for me.
if the feature exists how do I configure the ocaml installation properly ?

**Comment by mlantas** [comment, marked as answer]
Are you using the [Remote WSL](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl) extension? You should have access to your WSL opam switches once you do "Open Folder in WSL" in VSCode.

**Reply by OscarLahaie** [reply]
Yes it is working with WSL but not with Cygwin. How can I make it work for Cygwin ?

**Reply by mlantas** [reply]
Oh, I thought you meant Cygwin was working but WSL wasn't.


To get Cygwin working, install [OCaml for Windows](https://fdopen.github.io/opam-repository-mingw/) and make sure the `ocaml-env` program is in your Windows PATH.
I think with the default installation directory you will need to add `C:\OCaml64\usr\local\bin` to the PATH. It'll be a different directory if it was installed somewhere else.

**Reply by OscarLahaie** [reply]
![image](https://user-images.githubusercontent.com/25438086/165371618-b73ad2ab-223c-4291-986a-2fd35a043fc5.png)
Like this ?

**Reply by mlantas** [reply]
No, `ocaml-env` is a program inside the `C:\OCaml64\usr\local\bin` folder.
What I mean by making it accessible is adding it to the `Path` variable that is a few lines down.

If you open the `Path` variable, `C:\OCaml64\usr\local\bin` must be one of the lines there.

**Reply by OscarLahaie** [reply]
Thank you very much, it works!

**Comment by OscarLahaie** [comment]
I have another question :
Why I get the icons on Cygwin :
<img width="184" alt="image" src="https://user-images.githubusercontent.com/25438086/165373020-e60c503a-bcad-476d-87ff-55a7f9c51553.png">

But not on WSL :
![image](https://user-images.githubusercontent.com/25438086/165373418-5f2eeaba-4f22-4676-8905-2401ef4ae5ef.png)

**Reply by mlantas** [reply]
That might be a bug with the way we get `.svg` assets. Please make an issue.
